shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Delegation/Certificates.hs

```diff
@@ -124,12 +118,32 @@ isRetirePool _ = False
 
 newtype PoolDistr crypto = PoolDistr
   { unPoolDistr ::
-      ( Map
-          (KeyHash 'StakePool crypto)
-          (Rational, Hash crypto (VerKeyVRF crypto))
-      )
+      Map (KeyHash 'StakePool crypto) (IndividualPoolStake crypto)
+  }
+  deriving stock (Show, Eq)
+  deriving newtype (ToCBOR, FromCBOR, NFData, NoUnexpectedThunks, Relation)
+
+data IndividualPoolStake crypto = IndividualPoolStake
+  { individualPoolStake :: !Rational,
+    individualPoolStakeVrf :: !(Hash crypto (VerKeyVRF crypto))
   }
-  deriving (Show, Eq, ToCBOR, FromCBOR, NFData, NoUnexpectedThunks, Relation)
+  deriving stock (Show, Eq, Generic)
+  deriving anyclass (NFData, NoUnexpectedThunks)
+
+instance Crypto crypto => ToCBOR (IndividualPoolStake crypto) where
+  toCBOR (IndividualPoolStake stake vrf) =
+    mconcat
+      [ encodeListLen 2,
+        toCBOR stake,
+        toCBOR vrf
+      ]
+
+instance Crypto crypto => FromCBOR (IndividualPoolStake crypto) where
+  fromCBOR = do
+    enforceSize "IndividualPoolStake" 2
```

Comment:
We've dropped use of `enforceSize`, since it requires a canonical encoding. See e.g. https://github.com/input-output-hk/cardano-ledger-specs/pull/1693/files#diff-4530c51901096f749a3cab05d38d8df2L204 for what to use instad.

Reply:
Ah, will fix.